packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -315,6 +317,9 @@ class ServiceConnectionManager {
     }
 
     _connectionAvailableController.add(service);
+
+    // Clear the isolates cache:
+    isolateManager.clearIsolateIdCache();
```

Comment:
why is this needed?

Reply:
I was mostly worried about clearing the cached isolate value because it was being used in `onMainIsolateChanged` (which was called on start up). I've now split out the registration logic from `onMainIsolateChanged` (because that seems to be why we were calling it on start up, to register the isolate).  So now:

- on start up, we call `registerIsolate` with the cached isolate
- when the isolate changes, `onMainIsolateChanged` is called, which calls `registerIsolate` with the new isolate